File: frontend/lib/domain/repositories/comment_repository.dart
import '../entities/comment.dart';

abstract class CommentRepository {
  Future<List<Comment>> getNovelComments(String novelId);
  Future<List<Comment>> getChapterComments(String novelId, String chapterId);
  Future<Comment> createComment(String novelId, String? chapterId, String content);
  Future<Comment> updateComment(String novelId, String commentId, String content);
  Future<void> deleteComment(String novelId, String commentId);
} 

File: adminapp/src/domain/repositories/CommentRepository.ts
import { Comment } from '../entities/Comment';

export interface CommentRepository {
  getComments(): Promise<Comment[]>;
  getCommentById(id: string): Promise<Comment>;
  getCommentsByNovelId(novelId: string): Promise<Comment[]>;
  createComment(comment: Omit<Comment, 'id' | 'createdAt' | 'updatedAt'>): Promise<Comment>;
  updateComment(id: string, comment: Partial<Comment>): Promise<Comment>;
  deleteComment(id: string): Promise<void>;
  getReportedComments(): Promise<Comment[]>;
  getCommentStats(): Promise<{
    totalComments: number;
    reportedComments: number;
    totalLikes: number;
  }>;
} 